core/src/modules/statusReporters.ts

```diff
@@ -414,3 +413,143 @@ function mergeMetrics(metrics: StatusMetric[]) {
 
   return mergedMetrics;
 }
+
+export namespace FinalSummaryReporters {
+  const lastRunStart = new Date(api.bootTime);
+  export namespace Sources {
+    export interface SourceData {
+      name: string;
+      profilesCreated: number;
+      profilesImported: number;
+      importsCreated: number;
+      error: string;
+    }
+    export async function getData(): Promise<Array<SourceData>> {
```

Comment:
@evantahler We'd discussed typing returns inside a function vs. when the function is declared.  How might I do that in this case?

Reply:
You might be close already as what you are returning is a function of the already-typed:

```
      const sources: {        [id: string]: {          name: string;          profilesCreated: number;          profilesImported: number;          importsCreated: number;          error: string;        };      } = {};
```

You could re-work that to be something like:

```ts
const sources: {[id:string]: SourceData } = {}
```

which may help

Reply:
Were you unable to type the return values to the function didn't need to be typed?

Reply:
Noting that this summary will only include sources which had a run.